<launch>

    <!--TCP/IP通讯的参数-->
    <param name="/tcp/port" value="6666"/>
    <param name="/tcp/waitime" value="-1"/>        <!--客户端等待与服务器建立连接的时间。若为-1，则无限时间等待-->

    <!--设置相机的分辨率和帧率
        根据测试： 
            1280*720       ：10FPS  
            1024*768       ：10FPS
            800*600        ：20FPS
        height weight 可选择：
            16:9   3840*2160     1920*1080     1280*720    
            4:3    2592*1944     2048*1536     1600*1200
                   1280*960      1024*768      800*600
                   640*480       320*240    
        -->
    <param name="cam_conf/width" value="1280"/>    
    <param name="cam_conf/height" value="720"/>
    <param name="cam_conf/fps" value="10"/>

    <!--设置读取与保存图片的路径-->
    <param name="pic_save_path" value="$(find sap_as_server)/pic_save"/>
    <param name="pic_pro_path" value="$(find sap_as_server)/pic_provide"/>

    <!--启动四个相机的节点,两个输入参数分别为相机名称、相机编号
         相机名称可以自定义，相机编号切不可重复
    -->

    <node name="read_and_pub_front" pkg="sap_as_server" type="read_and_pub" output="screen" args="front usb 2"/>
    <node name="read_and_pub_back" pkg="sap_as_server" type="read_and_pub" output="screen" args="back usb 3"/>
    <node name="read_and_pub_left" pkg="sap_as_server" type="read_and_pub" output="screen" args="left csi 1"/>
    <node name="read_and_pub_right" pkg="sap_as_server" type="read_and_pub" output="screen" args="right csi 0"/>

    <!--启动图像获取与拼接节点-->
    <node name="pic_splice" pkg="sap_as_server" type="pic_splice" output="screen">
        <param name="cam_nums" value="4"/>          <!--打开相机的个数-->
        <param name="img_show" value="0"/>          <!--显示图片-->
    </node>

    <node name="server" pkg="sap_as_server" type="core_node" output="screen"/>

</launch>
